src/wazuh_modules/wm_gcp.c

```diff
@@ -0,0 +1,267 @@
+        wm_strcat(&command, "--project", ' ');
+        wm_strcat(&command, data->project_id, ' ');
+    }
+    if (data->subscription_name) {
+        wm_strcat(&command, "--subscription_id", ' ');
+        wm_strcat(&command, data->subscription_name, ' ');
+    }
+    if (data->credentials_file) {
+        wm_strcat(&command, "--credentials_file", ' ');
+        wm_strcat(&command, data->credentials_file, ' ');
+    }
+
+    if (data->max_messages) {
+        char *int_to_string;
+        os_malloc(OS_SIZE_1024, int_to_string);
+        sprintf(int_to_string, "%d", data->max_messages);
+        wm_strcat(&command, "--max_messages", ' ');
+        wm_strcat(&command, int_to_string, ' ');
+        os_free(int_to_string);
+    }
+    if (data->logging) {
+        char *int_to_string;
+        os_malloc(OS_SIZE_1024, int_to_string);
+        sprintf(int_to_string, "%d", data->logging);
+        wm_strcat(&command, "--log_level", ' ');
+        wm_strcat(&command, int_to_string, ' ');
+        os_free(int_to_string);
+    }
+
+    // Execute
+
+    mtdebug1(WM_GCP_LOGTAG, "Launching command: %s", command);
+
+    const int wm_exec_ret_code = wm_exec(command, &output, &status, 0, NULL);
+
+    os_free(command);
+
+    if (wm_exec_ret_code != 0){
+        mterror(WM_GCP_LOGTAG, "Internal error. Exiting...");
+        if (wm_exec_ret_code > 0) {
+            os_free(output);
+        }
+        pthread_exit(NULL);
+    }
+
+    if (status > 0) {
+        mtwarn(WM_GCP_LOGTAG, "Command returned exit code %d", status);
+        if(status == 1) {
+            char * unknown_error_msg = strstr(output,"Unknown error");
+            if (unknown_error_msg == NULL)
+                mtwarn(WM_GCP_LOGTAG, "Unknown error.");
+            else
+                mtwarn(WM_GCP_LOGTAG, "%s", unknown_error_msg);
+        } else if(status == 2) {
+            char * ptr;
+            if (ptr = strstr(output, "integration.py: error:"), ptr) {
+                ptr += 14;
+                mtwarn(WM_GCP_LOGTAG, "Error parsing arguments: %s", ptr);
+            } else {
+                mtwarn(WM_GCP_LOGTAG, "Error parsing arguments.");
+            }
+        } else {
+            char * ptr;
+            if (ptr = strstr(output, "ERROR: "), ptr) {
+                ptr += 7;
+                mtwarn(WM_GCP_LOGTAG, "%s", ptr);
+            } else {
+                mtwarn(WM_GCP_LOGTAG, "%s", output);
+            }
+        }
+        mtdebug1(WM_GCP_LOGTAG, "OUTPUT: %s", output);
+    } else {
+        mtdebug2(WM_GCP_LOGTAG, "OUTPUT: %s", output);
+    }
+
+    char *line;
+    char *save_ptr;
+
+    for (line = strtok_r(output, "\n", &save_ptr); line; line = strtok_r(NULL, "\n", &save_ptr)) {
+        char *msg_name;
+
+        switch (data->logging) {
+            case 0:
+                mtinfo(WM_GCP_LOGTAG, "Logging disabled.");
+                break;
+            case 1:
+                msg_name = strstr(line, "- DEBUG -");
+
+                if (msg_name) {
```

Comment:
We could change this validation to be made against `strstr(line, "- DEBUG -");` and remove msg_name entirely.